receiver/k8sclusterreceiver/metadata.yaml

```diff
@@ -245,6 +245,12 @@ metrics:
     unit: 1
     gauge:
       value_type: int
+  k8s.pod.status_reason:
+    enabled: false
+    description: Current status reason of the pod (1 - Evicted, 2 - NodeAffinity, 3 - NodeLost, 4 - Shutdown, 5 - UnexpectedAdmissionError, 6 - Unknown)
```

Comment:
Hi, is this an exhaustive list of reason values that could be set for a pod's state? If it is not, I fear we'll have to keep up with other possible reasons. 

Can you describe your use case for tracking the reason as metric? From what I have seen, the reason is rarely set on pod status. So for majority of the pods this metric will emit the value 6 for the entirety of the pod's lifecycle. Additional, some of the reasons are really just a no-op for the user. For e.g., when pod is evicted, k8s will bring up a replacement on another node. The reason `Shutdown`, I think is set when k8s evicts/terminates pod as part of graceful shutdown process of a node, again not much the user needs to do here for most cases because it will be replaced. For `NodeLost`, wouldn't it be better to track the node condition metric?

Reply:
@povilasv thanks for the update. I am not much concerned about this particular addition, but I do think we should be more intentional when introducing new metrics to the receiver, instead of replicating what kube-state-metrics (KSM) does. KSM does a good job of generating very detailed metrics for k8s objects already and is well-maintained, so it might be more appropriate to run KSM and use the prometheus receiver to collect from KSM if users are looking for same granularity of metrics.

Reply:
I agree with trying to represent KSM in otel natively. Running KSM and a collector to receive them feels unnecessary if the Collector could do it instead, plus we lose out on OTEL semantic conventions.  If KSM emitted OTLP and used otel semantic conventions I'd feel differently.

Reply:
@jinja2 can we call it resolved for this particular PR?

Reply:
Yeah, we can go ahead